datasource/ethereum/src/ethereum_adapter.rs

```diff
@@ -1007,7 +1007,17 @@ where
                 }
             }
             // Decode the return values according to the ABI
-            .and_then(move |output| call.function.decode_output(&output).map_err(From::from)),
+            .and_then(move |output| {
+                // We got a `0x` response. For Geth, this can mean a revert. It can also be that
```

Comment:
What do you think about moving this comment into the `if` branch? It being outside suggests that regardless of the check below, we already got `0x` response.

Reply:
True, moving it.